LogsHandler/src/test/java/bootTest/IntegrationTest.java

```diff
@@ -42,36 +36,110 @@ public static void beforeAllTests() {
+        assertTrue(401 == indexResponse.getStatus());
 
-        String result = postResponse.toString() + "\n" + postResponse.readEntity(String.class);
-        logger.debug(result);
+        await().atMost(Duration.ofSeconds(10)).untilAsserted(() -> searchWithInvalidToken(key, token));
+    }
 
+    @After
+    public void afterTest() {
+    }
+
+    private Account createAccount(CreateAccountRequest createAccountRequest){
+        Response response = webTarget.path("/create-account")
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.USER_AGENT, userAgent)
+                .header(HttpHeaders.CONTENT_TYPE, "application/json")
+                .post(Entity.json(createAccountRequest));
+        assertEquals(HttpURLConnection.HTTP_CREATED, response.getStatus());
+        Account account = response.readEntity(Account.class);
+        assertNotNull(account);
+        return account;
+    }
 
-        String header = "Macintosh";
-        await().atMost(Duration.ofSeconds(7)).until(() -> {
-            Response searchResponse1 = webTarget.path("search")
-                    .queryParam("message", key)
-                    .queryParam("header", header)
-                    .request(MediaType.APPLICATION_JSON)
-                    .get();
-            assertNotNull(searchResponse1);
-            String entity1 = searchResponse1.readEntity(String.class);
-            boolean isMessageIndexed1 = searchResponse1.getStatus() == HttpURLConnection.HTTP_OK;
-            boolean isMessageFound1 = entity1.indexOf(key) > -1;
+    private void indexAndAssert(String token, String message){
+        Response response = webTarget.path("/index")
+                .request(MediaType.APPLICATION_JSON)
+                .header(HttpHeaders.USER_AGENT, userAgent)
+                .header(X_ACCOUNT_TOKEN, token)
+                .post(Entity.json(message));
+        assertNotNull(response);
+        assertTrue(200 == response.getStatus());
+    }
 
-            return isMessageIndexed1 && isMessageFound1;
-        });
+    private void searchAndAssertAccountLogs(String key, String token) {
+        Response searchResponse = webTarget.path("search")
+                .request(MediaType.APPLICATION_JSON)
+                .header(this.X_ACCOUNT_TOKEN, token)
+                .get();
+        assertNotNull(searchResponse);
+        String entity = searchResponse.readEntity(String.class);
+        assertTrue(entity.indexOf(key) == -1);
+    }
 
+    private void searchAndAssertMessage(String key, String token) {
+        Response searchResponse = webTarget.path("search")
+                .request(MediaType.APPLICATION_JSON)
+                .header(this.X_ACCOUNT_TOKEN, token)
+                .get();
+        assertNotNull(searchResponse);
+        String entity = searchResponse.readEntity(String.class);
+        assertTrue(searchResponse.getStatus() == HttpURLConnection.HTTP_OK);
+        assertTrue(entity.indexOf(key) > -1);
```

Comment:
IntelliJ suggests you to change to:
```suggestion
        entity.contains(key)
```